README.md

```diff
@@ -78,11 +82,25 @@ You can also add some "augmenting" configs on top of the "core" config by extend
 This plugin provides the following core configurations:
 
 - [esnext](lib/config/esnext.js): Use this for anything written with ES2015+ features.
-- [react](lib/config/react.js): Use this for React projects.
+- [typescript](lib/config/typescript.js): Use this for Typescript projects. This configuration enables a subset of Typescript-specific that do not require type-checking to run. To enable all Typescript rules, you must augment this config with the `typescript-requiring-type-checking` config mentioned below.
 - [es5](lib/config/es5.js): Use this for legacy projects.
 
 This plugin also provides the following tool-specific configurations, which can be used on top of the core configurations:
 
+- [typescript-requiring-type-checking](lib/config/typescript-requiring-type-checking.js) Use this to augment the `typescript` config and enable all TypeScript rules, including those that require type checking. These rules are slower to run, and you will need to specify a path to your tsconfig.json file in the "project" property of "parserOptions". The following example would provide all of the TypeScript rules, assuming the tsconfig.json is in the same directory as your ESlint configuration.
```

Comment:
I would drop `-requiring` from the name

Reply:
Alternative naming bikeshedding: `typescript-extended` or `typescript-advanced`?